packages/react-server/src/ReactFlightServer.js

```diff
@@ -437,6 +515,19 @@ export function resolveModelToJSON(
     return null;
   }
 
+  if (
+    value.$$typeof === REACT_SERVER_CONTEXT_TYPE &&
+    key === '4' &&
+    parent[0] === REACT_PROVIDER_TYPE
+  ) {
+    popProvider((value: any));
+    if (__DEV__) {
+      insideContextProps = null;
+      isInsideContextValue = false;
+    }
+    return (undefined: any);
```

Comment:
Since this is inside an array, I believe this will be emitted as `null` since JSON emits null in place of undefined in arrays. That wastes bytes. Therefore, it would be better if this marker was part of an object instead since that would just remove the property from being encoded.

For example, you can make a clone of the props object and add it there.

Reply:
Sure, in this case I'll have to reserve a key to do this that can't be used as a prop key. `__popProvider$$` ?

Reply:
You know there's only two other keys so you can name it something shorter. `__pop`